### PR TITLE
Feature/multi speakers promo codes

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3270908&stafilter=NotComplete&fileview=grid

ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3270936&stafilter=NotComplete&fileview=grid

### CODE REVIEW

File: src/actions/speaker-actions.js [hunk 1]
```diff
@@ -873,8 +873,22 @@ export const sendSpeakerEmails = (currentFlowEvent,
         };
 
         if (promoCodeStrategy === 4) {
```
Comment (smarcet) [inline src/actions/speaker-actions.js:875]: lets use a constant here 4 seems to be a magic number

File: src/reducers/promocodes/promocode-specification-reducer.js [hunk 2]
```diff
@@ -46,8 +47,9 @@ const promocodeSpecificationReducer = (state = DEFAULT_STATE, action) => {
             return {...state,  entity: {...DEFAULT_ENTITY}, errors: {} };
         }
         case UPDATE_SPECS: {
-            const {entity} = payload;
-            return {...state,  entity: {...entity}, errors: {} };
+            const {promoCodeStrategy, entity} = payload;
+            const ticketTypes = promoCodeStrategy === 4 && entity.applyToAllTix ? [] : entity.ticketTypes;
```
[the remark this answers is not in the document]
Comment (romanetar) [reply src/reducers/promocodes/promocode-specification-reducer.js:51]: done

File: src/actions/speaker-actions.js (hunk 1 above)
Comment (romanetar) [reply src/actions/speaker-actions.js:875]: done

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**